src/stockpile.clj

```diff
@@ -118,12 +118,14 @@
             true))
         (throw ex)))))
 
-(defn- qpath [q]
-  (.resolve (:directory q) "q"))
+(defn- qpath ^Path [{:keys [^Path directory]}]
+  (.resolve directory "q"))
```

Comment:
Hmm.  If there's no reflection/perf difference, I might prefer the original formulation since it seems more obvious to me that the function expects a queue.